update_version.sh

```diff
@@ -1,31 +1,49 @@
 #!/usr/bin/env bash
 
-if [[ ("$2" == "poetry") || (("$2" == "") && ( -f pyproject.toml)) ]]; then
+function is_output_valid() {
+    line_numbers=$(eval "echo $1 | wc -l")
+    is_version_correct=$(eval "echo $1 | grep -E '[0-9.]?'")
+
+    if [[ ($line_numbers -eq 1)  && (! -z $is_version_correct)]]; then
+        return 0
+    else
+        return 1
+    fi
+
+}
+
+if [[ ("$1" == "poetry") ]]; then
   version_file="pyproject.toml"
-  command="poetry version -s"
-elif [[ ("$2" == "version_file") || (("$2" == "") && ( -f VERSION)) ]]; then
+  command="sed 's/[\ \"]*//g' | grep -E '(^\[tool\.poetry\])|(^version)' | sed -E 's/\[tool.poetry\]//g' | sed -E 's/version[[:blank:]]?=[[:blank:]]?//g'"
```

Comment:
please consider using functions instead of string command:
```shell
function cmd1() { echo "cmd1" }
function cmd2() { echo "cmd2" }

alias chosen_command='cmd1'
chosen_command  # calls cmd1

``

Reply:
also I think it is possible to pipe output of previous command as the input of the function, please check https://stackoverflow.com/a/11454477/1768976